test/Libraries/DynamoPythonTests/PythonEvalTestsWithLibraries.cs

```diff
@@ -196,17 +206,87 @@ import clr
 typedDictionary = DummyCollection.AcceptDictionary(d)
 typedDictionary['four'] = 4
 
-OUT = untypedDictionary, typedDictionary
+# Python dict => .NET IEnumerable - Returns keys in both engines
+sortedKeys = List.Sort(d)
+
+OUT = untypedDictionary, typedDictionary, sortedKeys
 ";
             var empty = new ArrayList();
-            var expected = new Dictionary<string, int> {
+            var expectedDictionary = new Dictionary<string, int> {
                 { "one", 1 }, { "two", 2 }, { "three", 3 }, { "four", 4 }
             };
+            var expectedKeys = new List<string> { "one", "three", "two" };
             var result = DSCPython.CPythonEvaluator.EvaluatePythonScript(code, empty, empty);
             Assert.IsTrue(result is IList);
-            foreach (var dict in result as IList)
+            var resultList = result as IList;
+            for (int i = 0; i < 2; i++)
+            {
+                Assert.IsTrue(resultList[i] is IDictionary);
+                DictionaryAssert(expectedDictionary, resultList[i] as IDictionary);
+            }
+            Assert.IsTrue(resultList[2] is IList);
+            CollectionAssert.AreEqual(expectedKeys, resultList[2] as IList);
+        }
+
+        [Test]
+        public void TestDictionaryViewsDecoding()
+        {
+            var code = @"
+import clr
+clr.AddReference('DSCoreNodes')
+from DSCore import List
+
+d = {'one': 1, 'two': 2, 'three': 3}
+
+dk = List.AddItemToEnd('four', d.keys())
+dv = List.AddItemToEnd(4, d.values())
+di = List.AddItemToEnd(('four', 4), d.items())
```

Comment:
Did we get all this supported by simply adding the `IsIterable`?

Reply:
Yes. They were all iterable, but we were checking if they were sequences, which they are not. Since our Python.NET code supported all iterables we are good!